llap-server/src/java/org/apache/hadoop/hive/llap/cache/MemoryManager.java

```diff
@@ -57,4 +57,13 @@
    *                        allocate the space.
    */
   void reserveMemory(long memoryToReserve, AtomicBoolean isStopped);
+
+  /**
+   * Ask the memory manager to evict more memory
+   *
+   * @param memoryToReserve amount of bytes to evict
+   * @return actual amount of evicted bytes.
+   */
+  long evictMemory(long memoryToReserve);
```

Comment:
Adding this to MemoryManager kind'o breaks the separation of memory manager and eviction policy (at least from interface perspective, these were two separate things so far).

Reply:
Memory manager is doing eviction on reserve call, thus this is adding an explicit handle to the eviction without reserving it. do you a different idea?